data-prepper-plugin-schema/src/test/java/org/opensearch/dataprepper/schemas/JsonSchemaConverterIT.java

```diff
@@ -59,6 +64,38 @@ void testSubTypes() throws JsonProcessingException {
         anyOfNode.forEach(aggregateActionNode -> assertThat(aggregateActionNode.has(PROPERTIES_KEY), is(true)));
     }
 
+    @Test
+    void test_examples() throws JsonProcessingException {
+        final ObjectNode jsonSchemaNode = objectUnderTest.convertIntoJsonSchema(
+                SchemaVersion.DRAFT_2020_12, OptionPreset.PLAIN_JSON, TestConfig.class);
+        assertThat(jsonSchemaNode, instanceOf(ObjectNode.class));
+        final JsonNode propertiesNode = jsonSchemaNode.at("/" + PROPERTIES_KEY);
+        assertThat(propertiesNode, instanceOf(ObjectNode.class));
+        assertThat(propertiesNode.has("string_value_with_two_examples"), is(true));
+        final JsonNode propertyNode = propertiesNode.at("/string_value_with_two_examples");
+        assertThat(propertyNode.has("type"), equalTo(true));
+        assertThat(propertyNode.get("type").getNodeType(), equalTo(JsonNodeType.STRING));
+
+        assertThat(propertyNode.has("examples"), equalTo(true));
+        assertThat(propertyNode.get("examples").getNodeType(), equalTo(JsonNodeType.ARRAY));
+        assertThat(propertyNode.get("examples").size(), equalTo(2));
+        assertThat(propertyNode.get("examples").get(0), notNullValue());
+        assertThat(propertyNode.get("examples").get(0).getNodeType(), equalTo(JsonNodeType.OBJECT));
+        assertThat(propertyNode.get("examples").get(0).has("example"), equalTo(true));
+        assertThat(propertyNode.get("examples").get(0).get("example").getNodeType(), equalTo(JsonNodeType.STRING));
+        assertThat(propertyNode.get("examples").get(0).get("example").textValue(), equalTo("some example value"));
+        assertThat(propertyNode.get("examples").get(0).has("description"), equalTo(false));
+
+        assertThat(propertyNode.get("examples").get(1), notNullValue());
+        assertThat(propertyNode.get("examples").get(1).getNodeType(), equalTo(JsonNodeType.OBJECT));
+        assertThat(propertyNode.get("examples").get(1).has("example"), equalTo(true));
+        assertThat(propertyNode.get("examples").get(1).get("example").getNodeType(), equalTo(JsonNodeType.STRING));
+        assertThat(propertyNode.get("examples").get(1).get("example").textValue(), equalTo("second example value"));
+        assertThat(propertyNode.get("examples").get(1).has("description"), equalTo(true));
+        assertThat(propertyNode.get("examples").get(1).get("description").getNodeType(), equalTo(JsonNodeType.STRING));
+        assertThat(propertyNode.get("examples").get(1).get("description").textValue(), equalTo("This is the second value."));
+    }
+
     @JsonClassDescription("test config")
     static class TestConfig {
         @JsonPropertyDescription("The aggregate action description")
```

Comment:
maybe make this description more generic

Reply:
This is just for testing.